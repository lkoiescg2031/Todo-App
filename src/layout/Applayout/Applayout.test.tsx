import React from "react";
import { render } from "@testing-library/react";

import AppLayout from "./Applayout";

describe("<AppLayout />", () => {
	it("matchShapshot test", () => {
		//arrange
		const titleName = "test";
		const utils = render(<AppLayout title={titleName} />);
		//action
		//assert
		expect(utils.container).toMatchSnapshot();
	});

	it("titlePropsTest", () => {
		//arrange
		const titleName = "title";
		const utils = render(<AppLayout title={titleName} />);
		//action
		//assert
		const title = utils.getByRole("heading"); // title 이 표시되었는 지 확인
		expect(title.children[0].textContent).toMatch(titleName);
		expect(title.children[1].textContent).toMatch(titleName);
	});
});
