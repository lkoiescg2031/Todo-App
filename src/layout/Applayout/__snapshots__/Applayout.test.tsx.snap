// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppLayout /> matchShapshot test 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h4
        class="MuiTypography-root AppLayout-title-3 MuiTypography-h4"
        role="heading"
      >
        <div
          class="PrivateHiddenCss-xsDown-6"
        >
          test
        </div>
        <div
          class="PrivateHiddenCss-smUp-8"
        >
          test
        </div>
      </h4>
    </div>
  </header>
  <div
    class="AppLayout-mainWrapper-1"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    />
    <main
      class="AppLayout-main-2"
    />
  </div>
</div>
`;
